src/battle_arcade_game_board.c

```diff
@@ -0,0 +1,1442 @@
+static void ov84_0223F8B4(BGCONFIG *config, WINDOW *window);
+static void ov84_0223F8E4(WINDOW *window);
+
+BOOL BattleArcadeGameBoard_InitOverlay(OVY_MANAGER *man, int *state) {
+    int i;
+    GAME_BOARD_WORK *work;
+    GAME_BOARD_ARGS *args;
+    
+    HandleLoadOverlay(FS_OVERLAY_ID(OVY_80), 2);
+    BattleArcadeGameBoard_InitSub();
+    CreateHeap(3, HEAP_ID_GAME_BOARD, 0x20000);
+    work = OverlayManager_CreateAndGetData(man, sizeof(GAME_BOARD_WORK), HEAP_ID_GAME_BOARD);
+    memset(work, 0, sizeof(GAME_BOARD_WORK));
+    
+    work->bgConfig = BgConfig_Alloc(HEAP_ID_GAME_BOARD);
+    work->man = man;
+    args = (GAME_BOARD_ARGS *) OverlayManager_GetArgs(man);
+    work->savedata = args->savedata;
+    work->arcadeSavedata = sub_02030E88(work->savedata);
+    work->arcadeScoreSavedata = sub_02030FA0(work->savedata);
+    work->type = args->type;
+    work->unk2A = args->unk1E;
+    work->winstreak = args->winstreak;
+    work->multiWinstreak = args->multiWinstreak;
+    work->unk12 = args->unk7;
+    work->returnWork = &args->returnWork;
+    work->options = Sav2_PlayerData_GetOptionsAddr(work->savedata);
+    work->playerParty = args->playerParty;
+    work->opponentParty = args->opponentParty;
+    work->unkE = 0xff;
+    work->frontierSavedata = sub_0203107C(work->savedata);
+    work->weather = args->weather;
+    work->cursorSpeed = args->cursorSpeed;
+    work->unk20 = args->unk10;
+    work->unk24 = args->unk14;
+    work->unk18 = 900;
+    work->randomFlag = args->randomFlag;
+    
+    for (i = 0; i < 32; i++) {
+        work->unk44[i] = 32;
+    }
+    
+    work->boardHeight = 4;
+    work->boardWidth = 4;
+    work->boardArea = 16;
+    
+    work->multiCursorPos = 0;
+    
+    *work->unk24 = BattleArcade_Random(work);
+    work->cursorPosStart = LCRandom() % 16;
+    
+    BattleArcadeGameBoard_InitObjects(work);
+    
+    if (BattleArcade_MultiplayerCheck(work->type) == TRUE) {
+        sub_02096910(work);
+    }
+    
+    *state = 0;
+    s_0223FA20[0] = 0;
+    s_0223FA20[1] = 0;
+    
+    return TRUE;
+}
+
+BOOL BattleArcadeGameBoard_Main(OVY_MANAGER *man, int *state) {
+    GAME_BOARD_WORK *work = OverlayManager_GetData(man);
+    
+    if (work->unkE != 0xff && *state == 1) {
+        ov84_0223F28C(work);
+        work->unk14 = ov84_0223EC88(work->boardArea, work->unkE);
+        BattleArcadeGameBoard_SetState(work, state, 2);
+    }
+    
+    switch (*state) {
+    case 0:
+        if (BattleArcadeGameBoard_Init(work) == TRUE) {
+            BattleArcadeGameBoard_SetState(work, state, 1);
+        }
+        break;
+    case 1:
+        if (BattleArcadeGameBoard_MoveCursor(work) == TRUE) {
+            if (work->unk2F_1 == 1) {
+                BattleArcadeGameBoard_SetState(work, state, 2);
+            } else {
+                if (BattleArcade_MultiplayerCheck(work->type) == TRUE) {
+                    BattleArcadeGameBoard_SetState(work, state, 3);
+                } else {    
```

Comment:
This can be an if, else if, else:
```
if (work->unk2F_1 == 1) {
  ...
} else if (BattleArcade_MultiplayerCheck(work->type) == TRUE) {
  ...
} else {
  ...
}
```

Reply:
Fixed, will be in the next push